msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountOverloadsMockedTest.java

```diff
@@ -86,10 +90,20 @@ public void setup() {
         super.setup();
         TestUtils.clearCache(SingleAccountPublicClientApplication.SINGLE_ACCOUNT_CREDENTIAL_SHARED_PREFERENCES);
         mSingleAccountPCA = (SingleAccountPublicClientApplication) mApplication;
-
-        MockHttpClient.reset();
-        MockHttpClient.setHttpResponse(MockServerResponse.getMockTokenSuccessResponse(), HttpClient.HttpMethod.POST, MockTokenCreator.MOCK_TOKEN_URL_REGEX);
-        MockHttpClient.setHttpResponse(MockServerResponse.getMockCloudDiscoveryResponse(), HttpClient.HttpMethod.GET, MockTokenCreator.CLOUD_DISCOVERY_ENDPOINT_REGEX);
+        mockHttpClient.intercept(
+                HttpRequestMatcher.builder()
+                        .method(m -> m.equals(HttpClient.HttpMethod.POST))
+                        .url(u -> u.toString().matches(MOCK_TOKEN_URL_REGEX))
```

Comment:
Since we're here, if these things are actually regular expressions, we should run Pattern.compile on them in the constants file and change this method.